xsd/bcsd.py

```diff
@@ -70,55 +66,94 @@ def make_source_grid(obj):
     return obj
 
 
-def bcsd(da_obs, da_train, da_predict, var='pr'):
+def _running_mean(obj, **kwargs):
+    '''helper function to apply rolling mean to groupby object'''
+    return obj.rolling(**kwargs).mean()
+
+
+def _regrid_to(dest, method='bilinear', *objs):
+    ''' helper function to handle regridding a batch of objects to a common
+    grid
+    '''
+    out = []
+    for obj in objs:
+        obj = _make_source_grid(obj)  # add grid info if needed
+        regridder = xe.Regridder(obj, dest, method)  # construct the regridder
+        out.append(regridder(obj))  # do the regrid op
+    return out
 
-    # add grid information to input arrays
-    da_obs = make_source_grid(da_obs)
-    da_train = make_source_grid(da_train)
-    da_predict = make_source_grid(da_predict)
+
+def bcsd(da_obs, da_train, da_predict, var='pr'):
+    ''' Apply the Bias Correction and Statistical Downscaling (BCSD) method.
+
+    Parameters
+    ----------
+    da_obs : xr.DataArray
+        Array representing the observed (truth) values.
+    da_train : xr.DataArray
+        Array representing the training data.
+    da_predict : xr.DataArray
+        Array representing the prediction data to be corrected using the BCSD
+        method.
```

Comment:
Note to self: all inputs are at the monthly timestep. This should be reflected in the doc string here.